pallets/pooled-staking/src/calls.rs

```diff
@@ -0,0 +1,473 @@
+            }
+            (SharesOrStake::Stake(s), TargetPool::ManualRewards) => {
+                pools::ManualRewards::<T>::stake_to_shares(&candidate, Stake(s))?.0
+            }
+        };
+
+        // Destroy shares
+        let removed_stake = match pool {
+            TargetPool::AutoCompounding => {
+                let stake = pools::AutoCompounding::<T>::sub_shares(
+                    &candidate,
+                    &delegator,
+                    Shares(shares),
+                )?;
+
+                if stake.0 > pools::AutoCompounding::<T>::hold(&candidate, &delegator).0 {
+                    Self::rebalance_hold(
+                        candidate.clone(),
+                        delegator.clone(),
+                        AllTargetPool::AutoCompounding,
+                    )?;
+                }
+
+                pools::AutoCompounding::<T>::decrease_hold(&candidate, &delegator, &stake)?;
+                stake
+            }
+            TargetPool::ManualRewards => {
+                let stake =
+                    pools::ManualRewards::<T>::sub_shares(&candidate, &delegator, Shares(shares))?;
+
+                if stake.0 > pools::ManualRewards::<T>::hold(&candidate, &delegator).0 {
+                    Self::rebalance_hold(
+                        candidate.clone(),
+                        delegator.clone(),
+                        AllTargetPool::ManualRewards,
+                    )?;
+                }
+
+                pools::ManualRewards::<T>::decrease_hold(&candidate, &delegator, &stake)?;
+                stake
+            }
+        };
+
+        // Create leaving shares.
+        // As with all pools there will be some rounding error, this amount
+        // should be small enough so that it is safe to directly release it
+        // in the delegator account.
+        let leaving_shares =
+            pools::Leaving::<T>::stake_to_shares_or_init(&candidate, removed_stake.clone())?;
+        let leaving_stake =
+            pools::Leaving::<T>::add_shares(&candidate, &delegator, leaving_shares.clone())?;
+        pools::Leaving::<T>::increase_hold(&candidate, &delegator, &leaving_stake)?;
+
+        // We create/mutate a request for leaving.
+        let now = frame_system::Pallet::<T>::block_number();
+        let operation_key = PendingOperationKey::Leaving {
+            candidate: candidate.clone(),
+            at_block: now,
+        };
+        let operation = PendingOperations::<T>::get(&delegator, &operation_key);
+        let operation = operation
+            .err_add(&leaving_shares.0)
+            .map_err(|_| Error::<T>::MathOverflow)?;
+        PendingOperations::<T>::set(&delegator, &operation_key, operation);
+
+        // We release the dust if non-zero.
+        let dust = removed_stake
+            .0
+            .err_sub(&leaving_stake.0)
+            .map_err(Error::<T>::from)?;
+
+        if dust.is_zero() {
```

Comment:
The comment above says

```
// We release the dust if non-zero.
```

So I guess this should be negated?

```suggestion
        if !dust.is_zero() {
```

If so, add a test.